pyexcel_text/__init__.py

```diff
@@ -112,6 +113,10 @@ def write_array(self, table):
         import tabulate
         if 'single_sheet_in_book' in self.keywords:
             self.keywords.pop('single_sheet_in_book')
+        if not isinstance(table, Matrix):
+            if not isinstance(table, list):
+                table = list(table)
```

Comment:
I assume this is because the incoming table is a generator, hence you need to treat it as normal sheet.